src/components/settings/settings.js

```diff
@@ -375,6 +376,9 @@ class Settings extends Component {
             ) : (
               <SmartContainerItem icon="speed" label={props.speedLabelText} options={speedOptions} onMenuChosen={this.onSpeedChange} />
             )}
+            {props.userSettingsMenus.map((menu, index) => (
+              <SmartContainerItem key={index} icon={menu.icon} label={menu.label} options={menu.options} onMenuChosen={menu.onMenuChosen} />
```

Comment:
what happens if one of the fields of menu is not set (e.g. options or label are missing) ? if some fields are optional - there should probably be a default for them, otherwise you should ignore a menu without values in all of the fields